api/service/anime.go

```
@@ -0,0 +1,262 @@
+		idValue = row
+	}
+
+	// open csv
+	f, err := os.Open("test.csv")
+	if err != nil {
+		log.Println("Unable to read test.csv", err)
+
+		return s
+	}
+
+	defer func(f *os.File) {
+		err := f.Close()
+		if err != nil {
+			log.Println("Unable to close test.csv", err)
+		}
+	}(f)
+
+	csvReader := csv.NewReader(f)
+	records, err := csvReader.ReadAll()
+	if err != nil {
+
+		return s
+	}
+	var newRecord []string
+	for i := 0; i < len(records); i++ {
+		value, _ := strconv.Atoi(records[i][0])
+		if value == idValue {
+			newRecord = records[i]
+
+			break
+		}
+	}
+
+	if id != "" && len(newRecord) == 0 {
+		log.Println("Record does not exists")
+
+		return s
+	}
+
+	animeID, err := strconv.Atoi(newRecord[0])
+	if err != nil {
+
+		return s
+	}
+
+	return animeI.AnimeStruct{
+		AnimeID:  animeID,
+		Title:    newRecord[1],
+		Synopsis: newRecord[2],
+		Studio:   newRecord[3],
+	}
+}
+
+func PostAnimeById(id string) int {
+	animeData := animeByIDExternalAPI(id)
+	animeValues := make([]string, 1)
+	for _, v := range structs.Values(animeData) {
+		animeValues = append(animeValues, fmt.Sprint(v))
+	}
+	// open csv
+	f, err := os.OpenFile("test.csv", os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+	if err != nil {
+		log.Println("Unable to open test.csv", err)
+
+		return http.StatusInternalServerError
+	}
+
+	defer func(f *os.File) {
+		err := f.Close()
+		if err != nil {
+			log.Println("Unable to close test.csv", err)
+		}
+	}(f)
+
+	csvwriter := csv.NewWriter(f)
+	defer csvwriter.Flush()
+
+	if err := csvwriter.Write(animeValues); err != nil {
+		log.Fatalln("error writing record to file", err)
+
+		return http.StatusInternalServerError
+	}
+
+	return http.StatusAccepted
+}
+
+func animeByIDExternalAPI(id string) animeI.AnimeStruct {
+	url := "https://api.jikan.moe/v3/anime/" + id
+	resp, err := http.Get(url)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	var result map[string]interface{}
+	json.NewDecoder(resp.Body).Decode(&result)
+	animeID, _ := result["mal_id"].(float64)
```

Never ignore errors. Fix here and all other occurrences of this issue.